s3auth-hosts/src/main/java/com/s3auth/hosts/DefaultDynamo.java

```diff
@@ -200,35 +205,74 @@ public void close() {
     @Cacheable.FlushBefore
     public boolean add(@NotNull final URN user,
         @NotNull final Domain domain) {
-        final ConcurrentMap<String, AttributeValue> attrs =
-            new ConcurrentHashMap<String, AttributeValue>(0);
-        attrs.put(DefaultDynamo.USER, new AttributeValue(user.toString()));
-        attrs.put(DefaultDynamo.NAME, new AttributeValue(domain.name()));
-        attrs.put(DefaultDynamo.KEY, new AttributeValue(domain.key()));
-        attrs.put(DefaultDynamo.SECRET, new AttributeValue(domain.secret()));
-        attrs.put(DefaultDynamo.REGION, new AttributeValue(domain.region()));
-        attrs.put(DefaultDynamo.SYSLOG, new AttributeValue(domain.syslog()));
-        attrs.put(DefaultDynamo.BUCKET, new AttributeValue(domain.bucket()));
-        final AmazonDynamoDB amazon = this.client.get();
-        amazon.putItem(new PutItemRequest(this.table, attrs));
-        amazon.shutdown();
-        return true;
+        final Region region = this.regionfactory.createRegion();
+        final Table curTable = region.table(this.table);
```

Comment:
Avoid composite variable names; something like `current` is sufficient. See http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html.